pandas/compat/_utils.py

```diff
@@ -0,0 +1,29 @@
+"""
+Utilities used in various `compat` components.
+"""
+
+from __future__ import annotations
+
+import pickle
+
+
+def flatten_buffer(b: bytes | bytearray | memoryview | pickle.PickleBuffer):
```

Comment:
Great change!

I'm not sure whether there is a policy about structuring `pandas.compat` @mroeschke - might be nicer to have this function and the two classes in one private file (or at least rename the files `lzma.py` and `bz2.py` to be private).

Reply:
Thanks!

Yeah happy to restructure however is easiest.

Just was running into cyclic `ImportError`s so had to start breaking things out somehow (even if it would be changed later).

Reply:
Yeah agreed with @twoertwein; would be nice to keep `flatten_buffer`, `BZ2File`, and `LZMAFile` together in one file so it's easier to see they are all related. Could still be a separate file like `pandas/compat/_pickle_compat.py` or similar.

Reply:
Could the output of this function be typed as well?

Reply:
Consolidated into `_compressors.py`. Happy to iterate further as needed